7581 merge vmxnet3s changes from open-vm-tools stable-10.0.x

dankimmel opened the pull request:
Reviewed by: Dan Kimmel <dan.kimmel@delphix.com>
Reviewed by: George Wilson <george.wilson@delphix.com>

The latest version of open-vm-tools has some of the same features
that we added to our version of vmxnet3s: support for jumbo-frames,
and some LSO fixes. We should merge these changes into our driver
(preferring the upstream version of the code in order to reduce
future merge pain).

Upstream bugs: DLPX-43227, DLPX-44800

dankimmel commented:
@zettabot go

dankimmel commented:
@zettabot go

sebroy commented on usr/src/uts/intel/io/vmxnet3s/vmxnet3_tx.c:
```diff
@@ -155,30 +126,32 @@ vmxnet3_tx_prepare_offload(vmxnet3_softc_t *dp, vmxnet3_offload_t *ol,
 			if (mblk != mp) {
 				ret = ol->hlen;
 			}
+		} else if (flags & HCK_PARTIALCKSUM) {
```
I'm assuming that vmxnet3 LSO also does the full hardware checksum, and that this is why VMware made this change. Given that we have no virtual hardware spec for vmxnet3, I can't be 100% sure.

rmustacc replied:
OK, makes sense. Thanks.

dankimmel commented:
I've updated this review to only include the changes from upstream, and posted the bug fixes around return values and memory use in a separate PR here: https://github.com/openzfs/openzfs/pull/239. I also fixed @rmustacc's comment about the incorrect `m_min_sdu`.

dankimmel commented:
@zettabot go

dankimmel commented:
@zettabot go

dankimmel commented:
The `dmu_tx_willuse_space` failure in the most recent `zloop` is a known, but unrelated, bug.
